gh-badge.spec.js

```diff
@@ -0,0 +1,61 @@
+var assert = require('assert');
+var cproc = require('child_process');
+var isPng = require('is-png');
+var isSvg = require('is-svg');
+
+describe('The CLI', function () {
+
+  it('should provide a help message', function(done) {
+    var child = cproc.spawn('node', ['gh-badge.js']);
+    var buffer = '';
+    child.stdout.on('data', function(chunk) {
+      buffer += ''+chunk;
+    });
+    child.stdout.on('end', function() {
+      assert(buffer.startsWith('Usage'));
+      done();
+    });
+  });
+
+  it('should produce default badges', function(done) {
+    var child = cproc.spawn('node',
+      ['gh-badge.js', 'cactus', 'grown']);
+    child.stdout.once('data', function(chunk) {
+      var buffer = ''+chunk;
```

Comment:
This is copied and pasted across all test cases - What about abstracting it out?
```js
function run(args, cb) {
  const child = spawn('node', ['gh-badge.js', ...args]);
  child.stdout.once('data', chunk => {
    const buffer = ''+chunk;
    cb(buffer);
  });
}
```

Then the test case would look like:
```js
it('should produce default badges', function(done) {
  run(['cactus', 'grown'], function(buffer) {
    assert.ok(isSvg(buffer));
    // .....
    done();
  });
});
```

Alternatively, instead of using `spawn`, what about splitting the logic from the CLI entry point (ie. have a function that takes `process.argv` and does the processing)? You could then simply `require` it and call it directly, without having to spawn separate Node.js processes.